tests/nodeos_read_terminate_at_block_test.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/env python3
+
+from testUtils import Utils
+from Cluster import Cluster
+from WalletMgr import WalletMgr
+from Node import Node
+from Node import ReturnType
+from TestHelper import TestHelper
+from testUtils import Account
+
+import re
+import os
+import time
+import signal
+import subprocess
+import shutil
+
+###############################################################
+# nodeos_read_terminate_at_block_test
+#
+# A few tests centered around read mode of irreversible, speculative and head with terminate-at-block set
+#
+###############################################################
+
+Print = Utils.Print
+errorExit = Utils.errorExit
+cmdError = Utils.cmdError
+relaunchTimeout = 10
+numOfProducers = 4
+totalNodes = 10
+
+termAtFutureBlockNum = 30
+
+# Parse command line arguments
+args = TestHelper.parse_args({"-v","--clean-run","--dump-error-details","--leave-running","--keep-logs"})
+Utils.Debug = args.v
+killAll=args.clean_run
+dumpErrorDetails=args.dump_error_details
+dontKill=args.leave_running
+killEosInstances=not dontKill
+killWallet=not dontKill
+keepLogs=args.keep_logs
+
+# Setup cluster and it's wallet manager
+walletMgr=WalletMgr(True)
+cluster=Cluster(walletd=True)
+cluster.setWalletMgr(walletMgr)
+
+# Wait for some time until LIB advance
+def waitForBlksProducedAndLibAdvanced():
+   requiredConfirmation = int(2 / 3 * numOfProducers) + 1
+   maxNumOfBlksReqToConfirmLib = (12 * requiredConfirmation - 1) * 2
+   # Give 6 seconds buffer time
+   bufferTime = 6
+   timeToWait = maxNumOfBlksReqToConfirmLib / 2 + bufferTime
+   time.sleep(timeToWait)
+
+# This function is special for "terminate-at-block" argument, if no this argument just use node.relaunch (see Node.py)
+def nodeRelaunch(node: Node, chainArg=None, newChain=False, skipGenesis=True, timeout=Utils.systemWaitTimeout, addSwapFlags=None, cachePopen=False, nodeosPath=None):
```

Comment:
I'm not sure I understand why are you using custom method here. regular `relaunch` has an option to add new arguments or swap existing. If any issue with that, I suggest just to update framework with this functionality.